src/charm.py

```diff
@@ -64,6 +66,19 @@ def _on_config_changed(self, event) -> None:
         self.model.unit.status = MaintenanceStatus(msg)
         self.configure_workload_container(event)
 
+    def _report_visits_by_ip_action(self, event: ActionEvent) -> None:
+        """Report requests to nginx by IP and report action result."""
+        container = self.unit.get_container(CONTAINER_NAME)
+        process = container.exec(
```

Comment:
What about running this check in Python? Just read the file and use a Python function? This could potentially make testing easier since we can easily test the Python function, testing a bash command tends to be more difficult (requires file creation etc)